### PR TITLE
Address warnings from log_loss metric

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Address warnings from log_loss metric
- `eps` arg is deprecated in sklearn.metrics.log_loss
- Have to renormalize during weighted ensemble fit to avoid warnings in the log_loss metric.

Mainline:

```
Fitting model: LightGBM ...
/opt/conda/envs/code310new/lib/python3.10/site-packages/sklearn/metrics/_classification.py:2851: FutureWarning: Setting the eps parameter is deprecated and will be removed in 1.5. Instead eps will always havea default value of `np.finfo(y_pred.dtype).eps`.
  warnings.warn(
/opt/conda/envs/code310new/lib/python3.10/site-packages/sklearn/metrics/_classification.py:2922: UserWarning: The y_pred values do not sum to one. Starting from 1.5 thiswill result in an error.
  warnings.warn(
	-1.1347	 = Validation score   (-log_loss)
	1.8s	 = Training   runtime
	0.0s	 = Validation runtime
```

This PR:

```
Fitting model: LightGBM ...
	-1.1347	 = Validation score   (-log_loss)
	1.84s	 = Training   runtime
	0.0s	 = Validation runtime
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by prateekdesai04 — APPROVED** [review]
LTGM
